### DISCUSSION TITLE
Add Tooltips explaining what the different styles do, maybe with images

### DISCUSSION THREAD

**Discussion by Kh4rj1** [lllyasviel/Fooocus#1723] [Ideas]
Currently in order to know what 'Fooocus V2' , 'Fooocus Enhance' , 'Fooocus Sharp' and 'Fooocus Negative' do, you have to find the correct documentation page here on github. I believe it would be better to have some sort of explanation in the web UI itself, maybe with a hover-over tooltip type of thing.

Also with the sheer amount of styles there are now, most people are never gonna try a lot of them out because there are just so many and you don't know what they do. I think it would be benficial to have 1-2 example images of every style available in the web UI, maybe make it so that you can toggle whether theres example images or not so it doesn't clog up the screen for those who dont want it. Otherwise it's just tedious to look through all the different styles not knowing what any of them do.

**Comment by docppp** [comment]
To be honest, I suspect that over 250 styles available in main GUI is far too many. I have no idea what most of them mean until I try them. I think that this is not great user experience. 

What I would like to see is maybe 20-30 main (the best lets say) styles for example chosen by community, and the rest would be on the page "More Styles" or something like that.

Idea of having some sort of show-up is also good. Maybe link "Show examples" at the top of the styles panel that opens static page with some images of styles generated with very basic prompt (like simple "cat"), so everybody can easily compare them. 

I could prepare something like that, but as the number of styles is really big right now, I need a confirmation first that this is going to be merged, so hundreds of images dont go to waste.

**Reply by Kh4rj1** [reply]
Totally agree, the current number styles just always on display is way too high and cumbersome

**Comment by Kh4rj1** [comment]
And while we're on the topic, I believe the various advanced image prompt variations could use some handy explainers as well. Like, to someone who works with this a lot it only makes sense that you select faceswap for the AI to use the image in that face as "inspiration"/ input for the new generation, but for me that wasn't immediately apparent at the start. All the other ones are even less understandable. Currently the way to figure this out is to search for the correct explanation somewhere on GitHub or maybe ask YouTube, but since the goal of this project is to have people focus on only prompting, some explanation of the different tools in the GUI itself would be beneficial imo.


Don't get me wrong, this is already by far the most "approachable" way to do advanced AI image generation, I just want to see it succeed which is why I'm posting my ideas for improvement :)

**Comment by lllyasviel** [comment]
We tried several image icons but they are kind of distracting and sometimes misleading.

Also, putting many colorful images on the UI will influence the color judgement of human eye perception, over generated images.

This is why most digital painting software use grey as bg color and try best to avoid putting colorful images on the UI.

**Reply by Kh4rj1** [reply]
True, so maybe an updated reference sheet could be linked somewhere

**Comment by docppp** [comment]
Please check this out:
https://github.com/lllyasviel/Fooocus/pull/1749

**Reply by Lola-Ice** [reply]
"Unable to render rich display"

**Comment by Nesuhst** [comment]
An idea I had while using the interface would be to show the actual prompt the style has on mouseover. So hover the mouse over the name of the style, and underneath the mouse, the text prompt for the positive and negative of that style would show up.

**Reply by mashb1t** [reply]
In general a good idea, but Gradio at least doesn't have a tooltip option by default, see https://github.com/lllyasviel/Fooocus/issues/1743#issuecomment-1877648595 for more information

**Comment by mashb1t** [comment]
fyi there is a new PR including preview thumbnails: https://github.com/lllyasviel/Fooocus/pull/1762
